libs/wingsdk/src/http/http.ts

```diff
@@ -0,0 +1,259 @@
+import { Code, InflightClient } from "../core";
+
+/**
+ * RequestCache-
+ * The cache read-only property that contains the cache mode of the request.
+ * It controls how the request will interact with the browser's HTTP cache.
+
+ */
+export enum RequestCache {
+  /**
+   * The browser looks for a matching request in its HTTP cache.
+   * If there is a match and it is fresh, it will be returned from the cache.
+   * If there is a match but it is stale, the browser will make a conditional request to the remote server.
+   * If the server indicates that the resource has not changed, it will be returned from the cache.
+   * Otherwise the resource will be downloaded from the server and the cache will be updated.
+   * If there is no match, the browser will make a normal request, and will update the cache with the downloaded resource.
+   */
+  DEFAULT = "default",
+  /**
+   * The browser fetches the resource from the remote server without first looking in the cache,
+   * and will not update the cache with the downloaded resource.
+   */
+  NO_STORE = "no-store",
+  /**
+   * The browser fetches the resource from the remote server without first looking in the cache,
+   * but then will update the cache with the downloaded resource.
+   */
+  RELOAD = "reload",
+  /**
+   * The browser looks for a matching request in its HTTP cache.
+   * If there is a match, fresh or stale, the browser will make a conditional request to the remote server.
+   * If the server indicates that the resource has not changed, it will be returned from the cache. Otherwise the resource will be downloaded from the server and the cache will be updated.
+   * If there is no match, the browser will make a normal request, and will update the cache with the downloaded resource.
+   */
+  NO_CACHE = "no-cache",
+  /**
+   * The browser looks for a matching request in its HTTP cache.
+   * If there is a match, fresh or stale, it will be returned from the cache.
+   * If there is no match, the browser will make a normal request, and will update the cache with the downloaded resource.
+   */
+  FORCE_CACHE = "force-cache",
+}
+/**
+ * The redirect read-only property that contains the mode for how redirects are handled.
+ */
+export enum RequestRedirect {
+  /**
+   * Follow all redirects incurred when fetching a resource.
+   */
+  FOLLOW = "follow",
+  /**
+   * Return a network error when a request is met with a redirect.
+   */
+  ERROR = "error",
+}
+/**
```

Comment:
Nit: add a little breathing room between each new class or enum (helps with readability, and in some editors like `vim` they make it easy for users to paginate through definitions if they are separated with newlines)
```suggestion
}

/**
```